### PR TITLE
Change tag name XML of the Camera 

### LINKED ISSUE #175: Change tag name XML of the Camera
This nodes:

```<RESOLUTIONS>```2560x1920```</RESOLUTIONS>```
replace by:
```<RESOLUTION>```2560x1920```</RESOLUTION>```

```<IMAGEFORMAT>```FLEX_RGBA_8888```</IMAGEFORMAT>```
replace by:
```<IMAGEFORMATS>```FLEX_RGBA_8888```</IMAGEFORMATS>```

```<SUPPORTS>```gc2355_qunhui_mipi_raw```</SUPPORTS>```
replace by:
```<SUPPORT>```gc2355_qunhui_mipi_raw```</SUPPORT>```

### PR DESCRIPTION
**Pull request by Ivans51**
Signed-off-by: Ivan Del Pino <idelpino@teclib.com>

### Changes description

Added a new way to read the list with the same name in the XML of output
Output XML:
```XML
    <CAMERAS>
      <RESOLUTION>2560x1920</RESOLUTION>
      <LENSFACING>BACK</LENSFACING>
      <FLASHUNIT>1</FLASHUNIT>
      <IMAGEFORMATS>JPEG</IMAGEFORMATS>
      <IMAGEFORMATS>YV12</IMAGEFORMATS>
      <IMAGEFORMATS>YUV_420_888</IMAGEFORMATS>
      <ORIENTATION>90</ORIENTATION>
      <FOCALLENGTH>3.5</FOCALLENGTH>
      <SENSORSIZE>4.29x3.19</SENSORSIZE>
      <MANUFACTURER>Hynix</MANUFACTURER>
      <RESOLUTIONVIDEO>1920x1088</RESOLUTIONVIDEO>
      <SUPPORTS>ov5670_mipi_raw</SUPPORTS>
      <SUPPORTS>hi551_mipi_raw</SUPPORTS>
      <SUPPORTS>gc2355_huaquan_mipi_raw</SUPPORTS>
      <SUPPORTS>gc2355_qunhui_mipi_raw</SUPPORTS>
      <MODEL>hi551mipiraw</MODEL>
    </CAMERAS>
```

Added a new way to read the list with the same name in the JSON of output
```JSON
"cameras": [
        {
          "resolution": "2560x1920",
          "lensfacing": "BACK",
          "flashunit": "1",
          "imageformats": "[JPEG, YV12, YUV_420_888]",
          "orientation": "90",
          "focallength": "3.5",
          "sensorsize": "4.29x3.19",
          "manufacturer": "Hynix",
          "resolutionvideo": "1920x1088",
          "supports": "[ov5670_mipi_raw, hi551_mipi_raw, gc2355_huaquan_mipi_raw, gc2355_qunhui_mipi_raw]",
          "model": "hi551mipiraw"
        },
        {
          "resolution": "2048x1152",
          "lensfacing": "FRONT",
          "flashunit": "0",
          "imageformats": "[JPEG, YV12, YUV_420_888]",
          "orientation": "270",
          "focallength": "3.5",
          "sensorsize": "4.99x2.55",
          "manufacturer": "GalaxyCore",
          "resolutionvideo": "N\/A",
          "supports": "[ov5670_mipi_raw, hi551_mipi_raw, gc2355_huaquan_mipi_raw, gc2355_qunhui_mipi_raw]",
          "model": "gc2355huaquanmipiraw;"
        }
      ]
```
### Checklist

Please check if your PR fulfills the following specifications:

- [x] Tests for the changes have been added
- [x] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@ivans |4|

### References

Closes: https://github.com/flyve-mdm/android-inventory-library/issues/175
Related #N/A
Depends on #N/A

### CODE REVIEW

**Review by ajsb85 — APPROVED** [review]

File: inventory/src/main/java/org/flyve/inventory/categories/Cameras.java [hunk 1]
```diff
@@ -445,19 +445,20 @@ public String getModel(int index) {
...
-    public String getSupportValue() {
+    public ArrayList<String> getSupportValue() {
         ArrayList<String> arrayList = new ArrayList<>();
         Iterator it = getCatInfoCamera("/system/lib/libcameracustom.so", "SENSOR_DRVNAME_", 100).iterator();
         while (it.hasNext()) {
             arrayList.add(((String) it.next()).toLowerCase(Locale.US));
         }
-        StringBuilder values = new StringBuilder();
+        return arrayList;
+        /*StringBuilder values = new StringBuilder();
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/categories/Cameras.java]: Remove the comment code

File: inventory/src/main/java/org/flyve/inventory/categories/Cameras.java [hunk 2]
```diff
@@ -445,19 +445,20 @@ public String getModel(int index) {
...
      */
-    public String getSupportValue() {
+    public ArrayList<String> getSupportValue() {
         ArrayList<String> arrayList = new ArrayList<>();
         Iterator it = getCatInfoCamera("/system/lib/libcameracustom.so", "SENSOR_DRVNAME_", 100).iterator();
         while (it.hasNext()) {
             arrayList.add(((String) it.next()).toLowerCase(Locale.US));
         }
-        StringBuilder values = new StringBuilder();
+        return arrayList;
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/categories/Cameras.java:457]: This method could return an empty array validate if the value is empty the tag should not appear on the xml or json

File: inventory/src/main/java/org/flyve/inventory/categories/Cameras.java [hunk 3]
```diff
@@ -445,19 +445,20 @@ public String getModel(int index) {
...
+        /*StringBuilder values = new StringBuilder();
         for (int i = 0; i < arrayList.size(); i++) {
             values.append(arrayList.get(i));
             if (i != arrayList.size() - 1)
                 values.append("; ");
         }
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/categories/Cameras.java]: Remove the comment code

File: inventory/src/main/java/org/flyve/inventory/categories/Category.java [hunk 4]
```diff
@@ -119,10 +120,17 @@ public String getTagName() {
...
+                   return super.put(key, value);
+               } else {
+                   return null;
+               }
            } else {
-               return null;
+               return super.put(key, value);
            }
        } else {
     	   return null;
```
Comment (rafaelje) [inline inventory/src/main/java/org/flyve/inventory/categories/Category.java:136]: possible NullPointerException

**Review by rafaelje — CHANGES_REQUESTED** [review]

File: inventory/src/main/java/org/flyve/inventory/categories/Category.java (hunk 4 above)
Comment (Ivans51) [reply inventory/src/main/java/org/flyve/inventory/categories/Category.java:136]: That method always return null if the condition no is true

File: inventory/src/main/java/org/flyve/inventory/categories/Cameras.java (hunk 2 above)
Comment (Ivans51) [reply inventory/src/main/java/org/flyve/inventory/categories/Cameras.java:457]: Validated in the last commit

**Review by rafaelje — APPROVED** [review]

**Pull request merged**